lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -332,9 +342,28 @@ defmodule Mix.Compilers.Elixir do
 
   ## Resolution
 
-  defp update_stale_sources(sources, changed) do
+  defp update_stale_sources(sources, changed, sources_stats \\ %{}) do
```

Comment:
It would be better to have two functions. One that receives the map and uses the first clause of the cond and another function that does exclusively the second clause of the cond.